app/javascript/Domain/ImperialGameCoordinator.js

```diff
@@ -117,6 +118,10 @@ export default class ImperialGameCoordinator {
       return;
     }
 
+    if (action.type !== 'undo') {
+      this.#game.addUndoCheckpoint();
```

Comment:
Just so I understand this, we add an empty array for every action and _some_ of those arrays will be populated with undo operations if the player did an undo on an action? But in most games most of those arrays will remain empty forever?

Reply:
In most of the time, the undo stack will be populated with operations, I haven't gotten around to implementing them yet since we'd want to keep the PRs small